model/src/cfd.rs

```diff
@@ -1470,6 +1478,38 @@ impl Cfd {
         })
     }
 
+    fn hours_to_extend_in_rollover_based_on_event(
+        &self,
+        candidate_event_id: BitMexPriceEventId,
+    ) -> Result<u64> {
+        let dlc = self.dlc.as_ref().context("Cannot roll over without DLC")?;
+        let current_settlement_time = dlc.settlement_event_id.timestamp();
+
+        let hours_left = current_settlement_time - OffsetDateTime::now_utc();
+
+        tracing::trace!(target = "cfd", time_left_in_cfd = %hours_left, "Calculating hours to extend in rollover");
+
+        if !hours_left.is_positive() {
+            tracing::warn!("Rolling over a contract that can be settled non-collaboratively");
+
+            return Ok(SETTLEMENT_INTERVAL.whole_hours() as u64);
+        }
```

Comment:
This should not happen right?

Reply:
I don't think so given how the application works.

But I don't think the rollover protocol should stop you if it happens.

Mind you this is just copied from the method introduced with version 2.